### DISCUSSION TITLE
Get URI of wallet connect through wagmi

### DISCUSSION THREAD

**Discussion by adeemfullstack909** [WalletConnect/walletconnect-monorepo#4171] [Developer Support] [answered]
How can i fetch walletconnect uri token through wagmi

My use case is this I'm trying to connect the metamask from a react ap with wagmi but on mobile it is opening the metamask in 2 steps like from the first page when i select wallet and click on connect button it opens walletConnect popup to select wallet again although in desktop its working fine and opening the metamask directly without that popup 
Same way i want to implement in mobile case 
Here is the code of connect wallet 

```
const connectToWallet = async (item) => {
        const wallet = {
            7: "injected",
            8: "coinbaseWallet",
        };
        let connectWallet = connectors.find((f) => f.id === wallet[item.id]);
        const isMobile = /iPhone|iPad|iPod|Android/i.test(navigator.userAgent);
        if (isMobile && !window.ethereum) {
            connectWallet = connectors.find((f) => f.id === "walletConnect");
        }
        if (!isConnected) {
            try {
                connect({ connector: connectWallet });
                localStorage.setItem("connectedWallet", connectWallet.name);
                setIsSignIn(true);
                // Now you are connected to MetaMask
            } catch (error) {
                console.error("Error connecting to wallet:", error);
                toastError(error);
            }
        } else {
            if (connectWallet.name === storedWallet) {
                signMessage();
            } else {
                toastError("Already connected to another wallet");
            }
        }
    };
```

Also here is the screenshot of popup which i dont want to show 
![File](https://github.com/WalletConnect/walletconnect-monorepo/assets/157484087/f9f74135-5829-4066-9785-5d231a0248cd)

So what i thought is if i can get the URI of wallet then it will not open this popup

**Comment by glitch-txs** [comment]
hi, this is an issue on the MetaMask side, if you do this on your own you'll get the same result
check here www.web3modal.com

**Reply by glitch-txs** [reply, marked as answer]
you can subscribe to the uri though: https://docs.walletconnect.com/advanced/providers/ethereum#use-without-walletconnectmodal

**Reply by ibrahimkhled** [reply]
،،،
